Kernel/Syscalls/open.cpp

```diff
@@ -54,7 +54,6 @@ ErrorOr<FlatPtr> Process::sys$open(Userspace<Syscall::SC_open_params const*> use
 
     dbgln_if(IO_DEBUG, "sys$open(dirfd={}, path='{}', options={}, mode={})", dirfd, path->view(), options, mode);
 
-    auto fd_allocation = TRY(allocate_fd());
```

Comment:
Why is this safer? `allocate_fd()` locks `m_fds` while making the allocation, and it's cleaned up if we return without using the allocation.